src/test/java/com/connexta/search/common/SearchTermITests.java

```diff
@@ -0,0 +1,120 @@
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+/**
+ * The purposes of this test class are to verify that all search terms can be added to index without
+ * causing an exception and that all search terms can be a part of a valid query sent to the index.
+ */
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+@Testcontainers
+@DirtiesContext
+public class SearchTermITests {
+
+  private static final int SOLR_PORT = 8983;
+
+  @Container
+  public static final GenericContainer solrContainer =
+      new GenericContainer("solr:8.1.1")
+          .withCommand("solr-create -c " + SOLR_COLLECTION)
+          .withExposedPorts(SOLR_PORT)
+          // TODO: would (new URL("solr", SOLR_COLLECTION, "admin", "ping").toString() work?
+          .waitingFor(Wait.forHttp("/solr/" + SOLR_COLLECTION + "/admin/ping"));
+
+  @Inject private QueryManager queryManager;
+  @Inject private IndexManager indexManager;
+  private Map<String, String> sampleDocument = createSampleIndexData();
+
+  public SearchTermITests() throws IOException {}
+
+  @NotNull
+  private Map<String, String> createSampleIndexData() throws IOException {
+
+    return Map.of(
+        ID_ATTRIBUTE_NAME,
+        UUID.randomUUID().toString(),
+        CONTENTS_ATTRIBUTE_NAME,
+        "Winterfell",
+        MEDIA_TYPE_ATTRIBUTE_NAME,
+        "application/json");
+  }
+
+  private String getQuery() {
+    return QUERY_TERMS.stream()
+        .map(term -> String.format("%s = '%s'", term, sampleDocument.get(term)))
+        .collect(Collectors.joining(" AND "));
+  }
+
+  InputStream getMetadata() throws IOException {
+    return IOUtils.toInputStream("{ \"ext.extracted.text\" : \"Winterfell\" }", "UTF-8");
+  }
+
+  @Test
+  void verifySampleDataComplete() {
+    assertThat(
+        "Sample document is not complete", QUERY_TERMS.equals(sampleDocument.keySet()), is(true));
+  }
+
+  @Test
+  public void test() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    Index document = mapper.convertValue(sampleDocument, Index.class);
```

Comment:
✏️ Why do you need to map this? `sampleDocument` is only used to hold pairs of terms and values, right? Would it be simpler to pass your tests values directly to indexManager#index?

Reply:
Good catch. That was left-over from some earlier code. I assumed an IndexManager's public method would accept an instance of Index. It did not. 

After looking through the IndexManagerImpl class it has two separate kinds of responsibilities:
1. Construct  an Index object from other objects (strings and a stream).
2. Talk to a Repository to determine if the item is allowed to be stored, and then storing it.

Responsibility is the builder pattern. It does not belong inside the manager class. The Index should be built outside the manager and passed into it.